pkg/kubelet/kubeletconfig/configsync_test.go

```diff
@@ -0,0 +1,283 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubeletconfig
+
+import (
+	"github.com/stretchr/testify/assert"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	fakeclientset "k8s.io/client-go/kubernetes/fake"
+	fakeEventv1 "k8s.io/client-go/kubernetes/typed/core/v1/fake"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/kubernetes/pkg/kubelet/kubeletconfig/checkpoint"
+	"k8s.io/kubernetes/pkg/kubelet/kubeletconfig/checkpoint/store"
+	"k8s.io/kubernetes/pkg/kubelet/kubeletconfig/status"
+	utilfs "k8s.io/kubernetes/pkg/util/filesystem"
+	"testing"
+)
+
+var (
+	nodeMeta        = metav1.ObjectMeta{Name: "mynode", UID: "mynode-uid"}
+	configMapA1Meta = metav1.ObjectMeta{Name: "foo", Namespace: "bar", UID: "uid-a1", ResourceVersion: "12345"}
+	configMapA2Meta = metav1.ObjectMeta{Name: "foo", Namespace: "bar", UID: "uid-a2", ResourceVersion: "54321"}
+)
+
+// The configuration has not changed, no need to call os.Exit(0)
+func TestSyncConfigSourceKubeletNoRestart(t *testing.T) {
+	tmpDir, err := utilfs.DefaultFs{}.TempDir("", "store-")
+	if err != nil {
+		t.Fatalf("can't make a temp dir: %v", err)
+	}
+	defer func() {
+		_ = utilfs.DefaultFs{}.RemoveAll(tmpDir)
+	}()
+
+	testCases := map[string]struct {
+		actualNCS    *apiv1.NodeConfigSource
+		assigned     bool
+		remoteConfig apiv1.ConfigMap
+	}{
+		"source, assigned is nil": {
+			actualNCS:    nil,
+			assigned:     false,
+			remoteConfig: apiv1.ConfigMap{},
+		},
+		"source is equal to assigned": {
+			actualNCS: &apiv1.NodeConfigSource{
+				ConfigMap: &apiv1.ConfigMapNodeConfigSource{
+					Namespace:       configMapA1Meta.Namespace,
+					Name:            configMapA1Meta.Name,
+					UID:             configMapA1Meta.UID,
+					ResourceVersion: configMapA1Meta.ResourceVersion,
+				}},
+			assigned: true,
+			remoteConfig: apiv1.ConfigMap{ObjectMeta: configMapA1Meta, Data: map[string]string{
+				"--v": "0",
+			}},
+		},
+	}
+
+	cc := &Controller{
```

Comment:
Thank you for your reply. I will try to modify it to keep the test sealed.
Since I use the local file system, I think I always need to clean up the local cache before the next test.

Reply:
done.
The thought of sealing test is great.  I got rid of `checkpointStore.Reset()`. :)